src/search.cpp

```diff
@@ -317,84 +141,245 @@ Search::iterator Search::begin() const {
         }
 
         if ( first ) {
-            this->valuesmap = read_valuesmap(database.get_metadata("valuesmap"));
-            language = database.get_metadata("language");
+            m_valuesmap = read_valuesmap(database.get_metadata("valuesmap"));
+            auto language = database.get_metadata("language");
             if (language.empty() ) {
-              // Database created before 2017/03 has no language metadata.
-              // However, term were stemmed anyway and we need to stem our
-              // search query the same the database was created.
-              // So we need a language, let's use the one of the zim.
-              // If zimfile has no language metadata, we can't do lot more here :/
-              try {
-                language = archive.getMetadata("Language");
-              } catch(...) {}
+                // Database created before 2017/03 has no language metadata.
+                // However, term were stemmed anyway and we need to stem our
+                // search query the same the database was created.
+                // So we need a language, let's use the one of the zim.
+                // If zimfile has no language metadata, we can't do lot more here :/
+                try {
+                    language = archive.getMetadata("Language");
+                } catch(...) {}
+            }
+            if (!language.empty()) {
+                icu::Locale languageLocale(language.c_str());
+                /* Configuring language base steemming */
+                try {
+                    Xapian::Stem stemmer = Xapian::Stem(languageLocale.getLanguage());
```

Comment:
If the overhead is not large, we should consider making stemmer a member variable so that it can be accessed by the  search_iterator. This will be useful for the implementation of #558.

Reply:
It would be nice yes. But it is a implementation improvement we can do later. Let's do this is another PR: